fast64_internal/oot/c_writer/oot_scene_writer/oot_scene_layer_to_c.py

```diff
@@ -0,0 +1,87 @@
+from ....utility import CData
+from ...oot_level_classes import OOTScene
+from ...oot_utility import indent
+
+from .oot_scene_cmds import ootSceneCommandsToC
+from ..oot_room_writer.oot_actor_to_c import ootActorListToC
+from ..oot_room_writer.oot_room_list_to_c import ootRoomListHeaderToC
+from ..oot_scene_writer.oot_path_to_c import ootPathListToC
+from ..oot_scene_writer.oot_light_to_c import ootLightSettingsToC
+from ..oot_scene_writer.oot_trans_actor_to_c import ootTransitionActorListToC
+from ..oot_scene_writer.oot_entrance_exit_to_c import ootEntranceListToC, ootExitListToC
+
+
+def ootGetSceneLayerData(scene: OOTScene, headerIndex: int):
+    """Returns a scene layer's data"""
+    layerData = CData()
+
+    # Write the start position list
+    if len(scene.startPositions) > 0:
+        layerData.append(ootActorListToC(scene, None, headerIndex))
+
+    # Write the transition actor list data
+    if len(scene.transitionActorList) > 0:
+        layerData.append(ootTransitionActorListToC(scene, headerIndex))
+
+    # Write the room segment list
+    if headerIndex == 0:
+        layerData.append(ootRoomListHeaderToC(scene))
+
+    # Write the path list
+    if len(scene.pathList) > 0:
+        layerData.append(ootPathListToC(scene))
```

Comment:
```suggestion
        # Write the path list
        if len(scene.pathList) > 0:
            layerData.append(ootPathListToC(scene))
```
Technically paths can be header specific, but for now they should only be written once.

Reply:
welp, now it's using the layer index for the array names so this shouldn't matter I guess? but since there's no layer choice in the UI I'll add this change 👍